### PR TITLE
[D&D] Adds visualization type switcher for Wizard

### LINKED ISSUE #954: [D&D] Visualization type selector
Add a visualization type selector above the canvas on the workspace

Screen Shot 2021-11-17 at 2 23 06 AM

### PR DESCRIPTION
**Pull request by ashwin-pc**
Signed-off-by: Ashwin Pc <ashwinpc@amazon.com>

### Description
- Adds a working Visualization Type picker for Wizard

Screen Shot 2022-08-29 at 3 53 40 PM

- Updates Wizard UI to match the latest mocks
image

### Issues Resolved
#954 
 
### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass
    - [x] `yarn test:jest`
    - [x] `yarn test:jest_integration`
    - [x] `yarn test:ftr`
- [ ] New functionality has been documented.
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by AMoo-Miki — APPROVED** [review]

File: src/plugins/wizard/public/application/components/side_nav.scss [hunk 1]
```diff
@@ -25,6 +46,5 @@
 }
 
 .wizDatasourceSelect {
-  max-width: $wizSideNavWidth;
-  padding: $euiSize $euiSize 0 $euiSize;
+  max-width: calc(#{$wizSideNavWidth} - 1px);
```
Comment (joshuarrrr) [inline src/plugins/wizard/public/application/components/side_nav.scss:49]: Should we track this as an enhancement we need in OUI?

File: src/plugins/wizard/public/application/components/data_source_select.tsx [hunk 2]
```diff
@@ -29,6 +29,7 @@ export const DataSourceSelect = () => {
   const { indexPatterns, loading, error, selected } = useIndexPatterns();
   const dispatch = useTypedDispatch();
 
+  // TODO: Should be a standard EUI component
```
Comment (kavilla) [inline src/plugins/wizard/public/application/components/data_source_select.tsx:32]: NIT: do we have a follow-up up issue? Doesn't have to be committed but so we can reference in the pull request?

File: src/plugins/wizard/public/visualizations/metric/components/metric_viz_options.tsx [hunk 3]
```diff
@@ -5,8 +5,7 @@
 
 import React, { useCallback } from 'react';
 import { i18n } from '@osd/i18n';
-import { EuiButtonGroup, EuiFormRow, EuiPanel, EuiSpacer, EuiTitle } from '@elastic/eui';
-import { FormattedMessage } from 'react-intl';
+import { EuiButtonGroup, EuiFormRow } from '@elastic/eui';
```
Comment (kavilla) [inline src/plugins/wizard/public/visualizations/metric/components/metric_viz_options.tsx:8]: can we explicitly use oui now? both should work right?

**Review by joshuarrrr — APPROVED** [review]
Cool, it'll be nice to have these changes!

**Review by kavilla — COMMENTED** [review]
Is size a factor? For me on my small screen the middle section gets small and I can't resize

**Comment by kavilla** [comment]
Screen Shot 2022-08-31 at 4 51 43 PM

File: src/plugins/wizard/public/application/components/data_source_select.tsx (hunk 2 above)
Comment (ashwin-pc) [reply src/plugins/wizard/public/application/components/data_source_select.tsx:32]: https://github.com/opensearch-project/oui/issues/51

File: src/plugins/wizard/public/visualizations/metric/components/metric_viz_options.tsx (hunk 3 above)
Comment (ashwin-pc) [reply src/plugins/wizard/public/visualizations/metric/components/metric_viz_options.tsx:8]: I tried that but Typescript was complaining. So i moved the issue out for later

File: src/plugins/wizard/public/application/components/side_nav.scss (hunk 1 above)
Comment (ashwin-pc) [reply src/plugins/wizard/public/application/components/side_nav.scss:49]: This is a specific with we need for the side panels. I've opened an issue to add resizing and collapsing https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2241. That should move this from CSS to JS

**Comment by ashwin-pc** [comment]
@kavilla This is a specific with we need for the side panels. I will narrow down the right side panel in the followup PR and I've opened an issue to add resizing and collapsing https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2241. That should mitigate this issue for smaller viewports.

The smaller right side panel is for the followup Barchart PR and the resize panel PR will come after that.

**Pull request merged**